### DISCUSSION TITLE
Getting a field value from database in relation manager

### DISCUSSION THREAD

**Discussion by haifan420** [filamentphp/filament#9235] [Help] [answered]
So I have this relation manager, each Babies has many Visits

![image](https://github.com/filamentphp/filament/assets/117128195/75c69fa8-e934-4315-95e9-7f43a8f2df3f)

In New Visit form, there is a `number_of_visit` field (basically indicating that the visit at this time is the X th visit)

![Screenshot_20231021_210506](https://github.com/filamentphp/filament/assets/117128195/f50c67e6-d9bd-40dd-9328-eaa6f39f59f5)
```
Forms\Components\TextInput::make('kunjungan')
                     ->label('Number of visits')
                     ->default(function () {
                              return \App\Models\Visit::query()->where('baby_id', 1)->count() + 1; 
                     }),
```

Instead of using constant value `'baby_id' = 1`, I want to display the number of visits for a specific baby dynamically based on the current baby's `id` (in this case baby named `Aquila`) so my question is how do i get the baby's `id` value for this relation manager. 

I hope this isn't confusing. If it is, ignore everything above, this is the query that I'm trying to achieve, maybe it could help
```
SELECT COUNT(*) FROM `babies`, `visits` 
WHERE babies.id = 1 // replace 1 with id of the baby that I'm currently viewing, I don't know how to get the `id` and put it in relation manager's form
AND babies.id = visits.baby_id;
```

**Comment by leandrocfe** [comment]
Do you have a visits relationship?

```php
TextInput::make('kunjungan')
    ->label('Number of visits')
    ->formatStateUsing(fn ($record): ?string => $record?->visits->count() ?? 0)
    ->disabled()
```

**Reply by haifan420** [reply]
Sorry, it seems that my question was a little confusing, what i want is to access the record of the `baby`,  the parent model of `visit`. Thank you for helping regardless.

**Comment by haifan420** [comment, marked as answer]
Found the solution from [this Laracasts post](https://laracasts.com/discuss/channels/code-review/filamentphp-how-to-access-model-record-from-relationshipmanager)
```
use Livewire\Component as Livewire;

Forms\Components\TextInput::make('kunjungan')
          ->label('Number of visits')
          ->default(
              function () {
                    function (Livewire $livewire) {
                         $livewire->ownerRecord;
                    };
              return \App\Models\Visit::query()->where('baby_id', $this->ownerRecord->id)->count() + 1;
              }
          ),
```

**Reply by leandrocfe** [reply]
ahh ok, I hadn't got the idea.. A shorter way

```php
Forms\Components\TextInput::make('kunjungan')
          ->label('Number of visits')
          ->default(fn ($livewire) => $livewire->ownerRecord->visits->count() + 1)
```

**Reply by haifan420** [reply]
Nice!

**Comment by castorius31** [comment]
Anothers suggestions when it depends on the context (ownerRecord can not exist)

````php
->default(function ($livewire) {
        if (isset($livewire->ownerRecord)) {
            return $livewire->ownerRecord->id;
        }
    }
)
->disabled(function ($livewire) {
    if (isset($livewire->ownerRecord)) {
        return true;
    }
    return false;
})
```
